### PR TITLE
CLN: remove _from_mgr (no longer used after GH-45363)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
One part of the follow-up to https://github.com/pandas-dev/pandas/pull/45363

This one doesn't necessarily need to be backported to 1.4

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -285,22 +285,6 @@ def _init_mgr(
                 mgr = mgr.astype(dtype=dtype)
         return mgr
 
-    @classmethod
-    def _from_mgr(cls, mgr: Manager):
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py:289]: Removed this, because it was only used in the groupby Splitter (and so that usage was removed in the previous PR)

**Comment by jreback** [comment]
we just merged some cleanup similar to this by @jbrockmendel so if you can rebase

**Comment by jorisvandenbossche** [comment]
@jbrockmendel it would be useful if you can mention the PR when doing a follow-up to a specific PR (so that github shows it as a linked PR in the original PR)

**Comment by jreback** [comment]
thanks @jorisvandenbossche

**Pull request merged**